app/services/gitlab/fetch_branches_service.rb

```diff
@@ -0,0 +1,33 @@
+module Gitlab
+  class FetchBranchesService
+    delegate :error_message, to: :api, allow_nil: true
+
+    def initialize(project)
+      @project = project
+    end
+
+    def fetch
+      @api ||= GitlabApi::FetchBranches.perform(
+        gitlab_integration.data['url'],
```

Comment:
gitlab_integrations is using find_by method, which returns  nil if there's no integration with the gitlab key, if is intended pls add a guard clause, if is not add find_by! and a proper error handling

Reply:
What would be a proper way to handle an error in this scenario?